hazelcast/src/main/java/com/hazelcast/client/impl/spi/ProxyManager.java

```diff
@@ -407,6 +409,22 @@ private void createCachesOnCluster() {
         }
     }
 
+    @Override
+    public void distributedObjectCreated(DistributedObjectEvent event) {
+        // no-op
+    }
+
+    @Override
+    public void distributedObjectDestroyed(DistributedObjectEvent event) {
+        if (event.getSource().equals(client.getConnectionManager().getClientUuid())) {
+            // ignore destroy events which were originally initiated from this client
+            return;
+        }
+
+        // destroy local proxy
+        destroyProxyLocally(event.getServiceName(), (String) event.getObjectName());
```

Comment:
just to note, local proxy can still be destroyed if client restarts(uuid changes) and event comes afterwards with previous uuid.

Reply:
Yes, this solution is not solving perfectly all cases. We are aware of this. We depend on events which are not reliable anyway. A better solution may be using version numbers for proxylist snapshots as done for membership and partition lists. This needs to be solved in future releases and there is an open PRD for redesigning the proxy create and destroy logics. This PR only aims to fix the related issue as best it can without a bigger design change.

Reply:
maybe we can note that this is a best effort solution and some known issues still exist.

Reply:
Noted.